pkg/cluster/context.go

```diff
@@ -100,8 +100,26 @@ func (c *Context) ClusterName() string {
 	return c.Name()
 }
 
+//CreateOption is a Context configuration option supplied to create
+type CreateOption func(*create.Context)
+
+// Retain configures create to retain nodes after failing for debugging pourposes
+func Retain(retain bool) CreateOption {
+	return func(c *create.Context) {
+		c.Retain = retain
```

Comment:
normally these would not be exported, so you can ensure that they are only set when applying the options. We can refactor further in a follow-up though